app/models/web_hook/event.rb

```diff
@@ -39,7 +39,7 @@ def model
 
     delegate :logger, to: :Rails
 
-    def has_transactional_callbacks?
+    def trigger_transactional_callbacks?
```

Comment:
What's the reason for this change?...

Reply:
I've been reading a bit and it seems to be required to run callbacks after a DB transaction succeeded or rollbacked.

In rails 6, `trigger_transactional_callbacks?` is expected to be true in order to actually call the callbacks. See [#36190](https://github.com/rails/rails/pull/36190/files#diff-dd0ea9dfea1fd486f2b1b5eac8c798f839467173a1c2315d3ad04f4098f7dda3).

The two files affected by this change are [/app/models/web_hook/event.rb](https://github.com/3scale/porta/blob/00d75d0f5e9a8cb936632243ca330c094488a0e6/app/models/web_hook/event.rb) and [/app/subscribers/after_commit_subscriber.rb](https://github.com/3scale/porta/blob/632667de4bade6a0cf2859341499d9a9363d68a9/app/subscribers/after_commit_subscriber.rb). You can compare the structure of these files with the one showed at this article:

https://www.dmitry-ishkov.com/2019/04/running-activejob-after-transaction.html

I don't really know how to verify this behavior, other than running the tests for the affected files: [/test/unit/web_hook/event_test.rb](https://github.com/3scale/porta/blob/e8feceb1e8950912b88d1273d3b1e5aeb4ac5fc1/test/unit/web_hook/event_test.rb) and [/test/subscribers/after_commit_subscriber_test.rb](https://github.com/3scale/porta/blob/1a48b303b61f4893e7eaa002dd4cf73f4474ab6d/test/subscribers/after_commit_subscriber_test.rb)

If we run these tests without the change, they fail with this error:

`"Minitest::UnexpectedError: NoMethodError: undefined method 'trigger_transactional_callbacks?'"`

After the change, the tests pass. I think we can assume this is correct, what do you think?

Reply:
I looked into it, and while I still do not understand completely these ActiveRecord internals, I think it makes sense.
The code is trying to make the classes that are not AR classes act like AR classes, e.g. only do things when the transaction was committed, and not otherwise. Like do not send a web hook if the transaction rolled back for some reason.